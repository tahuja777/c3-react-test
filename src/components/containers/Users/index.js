import React, { Component } from "react";
import User from '../../presentation/User';

class Users extends Component {
  constructor(props) {
    super(props);

    this.state = {
      users: [
        {
          id: 4,
          first_name: "Eve",
          last_name: "Holt",
          avatar:
            "https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
        },
        {
          id: 5,
          first_name: "Charles",
          last_name: "Morris",
          avatar:
            "https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
        },
        {
          id: 6,
          first_name: "Tracey",
          last_name: "Ramos",
          avatar:
            "https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
        }
      ]
    };
  }

  getUsersFromApi = () => {
    
    fetch('https://reqres.in/api/users?page=1&per_page=10')
    .then(response => 
        response.json().then(data => ({
            data: data,
            status: response.status
        })
    ).then(res => {
        
          let { data } = res.data;
          if(data.length > 0){
            // @existingUsersArr holds the users id already present in state
            let existingUsersArr = this.state.users.map( (user) => {
                return user.id
            });
      
            // @newUsersList holds user ids which are not already present in state
            let newUsersList = data.filter( (user) => {
              return (!existingUsersArr.includes(user.id))
            });

            // @updatedUsers holds unique users and no duplicates
            let updatedUsers = [...this.state.users, ...newUsersList];
            
            // updating users list in state after filtering duplicate
            this.setState({
                users: updatedUsers
            })
          }
        
    }))
    .catch( error => console.error(error));
  }

  componentDidMount(){
      // Call method to fetch users list and change state data
      this.getUsersFromApi(); 
  }

  deleteUserHandler = (e, id) => {
    e.preventDefault();
    let usersList = [...this.state.users];
    // @updatedUsersList holds the users list after filtering deleted user
    let updatedUsersList = usersList.filter( (user) => {
        return user.id !== id;
    });

    // updating state after deleting user
    this.setState({
        users: updatedUsersList
    })
  }

  render() {
    let usersList = null;
    if(this.state.users.length > 0){
      usersList = this.state.users.map( (user) => {
          return <User 
                    key={user.id} 
                    userId={user.id}
                    firstName={user.first_name} 
                    lastName={user.last_name}
                    profileAvatar={user.avatar}
                    deleteUser={(e, id) => this.deleteUserHandler(e, id)} /> 
      });
    }

    return (
      <div className="pearon-users">
        <h1>Pearson User Management</h1>
        <div className="users-list-box">
          <div className="inner-user-box">
            <ul className="users-box">
              { /* When there will be no users then message will be displayed "No users found!". This situation would arise when we delete all users by clicking delete. */}
              {this.state.users.length > 0 ? usersList : <div className="no-user-msg">No users found!</div>}
            </ul>
          </div>
        </div>    
      </div>
    );
  }
}
export default Users;