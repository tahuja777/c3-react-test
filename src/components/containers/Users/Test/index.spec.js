import React from "react";
import { shallow, mount } from "enzyme";
import Users from "../index";

describe("Users", () => {
  let usersWrapperMount;
  let usersWrapperShallow;

  beforeEach(() => {
    //Mock fetch API used for getting users list.
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
        json: function() {
          let mockObj = {
                          data : [
                                    {
                                      id: 6,
                                      first_name: "Eve1",
                                      last_name: "Holt1",
                                      avatar:
                                        "https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                                    }
                                  ],
                          };
          return Promise.resolve(mockObj)
        },
        status: 200}));
        usersWrapperShallow = shallow(<Users />);
        usersWrapperMount = mount(<Users />);
  });

  test('calls componentDidMount', () => {
      const spy = jest.spyOn(Users.prototype, 'componentDidMount');
      usersWrapperShallow.instance().componentDidMount();
      expect(spy).toHaveBeenCalled();
  });

  test("should render with required props", () => {
    expect(usersWrapperShallow).toMatchSnapshot();
  })

  test("renders a h1", () => {
    const h1 = usersWrapperShallow.find("h1");
    expect(h1.text()).toEqual("Pearson User Management");
  });

  test("should delete user on click", () => {
    const deleteUserHandlerSpy = jest.spyOn(usersWrapperMount.instance(), "deleteUserHandler");
    usersWrapperMount.find('.delete-user').first().simulate('click');
    expect(usersWrapperMount.state().users.length).toEqual(2);
  });

  test("renders message when no users present", () => {
    usersWrapperShallow.setState({ users: [] });
    const div = usersWrapperShallow.find(".no-user-msg");
    expect(div.text()).toEqual("No users found!");
  });
 
  test("renders users first and last name", () => {
      const userName = usersWrapperMount.find(".user-name");
      expect(userName.length).toEqual(3);
  });

  test("renders delete user link", () => {
      const deletelink = usersWrapperMount.find(".delete-user");
      expect(deletelink.length).toEqual(3);
  });
  
  test("renders users avatar", () => {
      const userAvatar = usersWrapperMount.find(".user-avatar");
      expect(userAvatar.length).toEqual(3);
  });

});

describe("Users API Error", () => {
  let usersWrapper;

  beforeEach(() => {
    window.fetch = jest.fn().mockImplementation(() => Promise.reject({
        json: function() {
          let mockObj = {
                          data : null
                          }
          return Promise.resolve(mockObj)
        },
        status: 404}));
        usersWrapper = shallow(<Users />);   
    
  });

  test("should not render when problem with API", () => {
    expect(usersWrapper).toMatchSnapshot();
  });
});
