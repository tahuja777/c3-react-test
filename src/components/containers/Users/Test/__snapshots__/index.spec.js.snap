// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users API Error should have no impact when problem with API 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Users />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Pearson User Management
</h1>,
        <div
          className="users-list-box"
>
          <div
                    className="inner-user-box"
          >
                    <ul
                              className="users-box"
                    >
                              <user
                                        deleteUser={[Function]}
                                        firstName="Eve"
                                        lastName="Holt"
                                        profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                                        userId={4}
                              />
                              <user
                                        deleteUser={[Function]}
                                        firstName="Charles"
                                        lastName="Morris"
                                        profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                                        userId={5}
                              />
                              <user
                                        deleteUser={[Function]}
                                        firstName="Tracey"
                                        lastName="Ramos"
                                        profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                                        userId={6}
                              />
                    </ul>
          </div>
</div>,
      ],
      "className": "pearon-users",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Pearson User Management",
        },
        "ref": null,
        "rendered": "Pearson User Management",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="inner-user-box"
>
            <ul
                        className="users-box"
            >
                        <user
                                    deleteUser={[Function]}
                                    firstName="Eve"
                                    lastName="Holt"
                                    profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                                    userId={4}
                        />
                        <user
                                    deleteUser={[Function]}
                                    firstName="Charles"
                                    lastName="Morris"
                                    profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                                    userId={5}
                        />
                        <user
                                    deleteUser={[Function]}
                                    firstName="Tracey"
                                    lastName="Ramos"
                                    profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                                    userId={6}
                        />
            </ul>
</div>,
          "className": "users-list-box",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="users-box"
>
              <user
                            deleteUser={[Function]}
                            firstName="Eve"
                            lastName="Holt"
                            profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                            userId={4}
              />
              <user
                            deleteUser={[Function]}
                            firstName="Charles"
                            lastName="Morris"
                            profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                            userId={5}
              />
              <user
                            deleteUser={[Function]}
                            firstName="Tracey"
                            lastName="Ramos"
                            profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                            userId={6}
              />
</ul>,
            "className": "inner-user-box",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <user
                  deleteUser={[Function]}
                  firstName="Eve"
                  lastName="Holt"
                  profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                  userId={4}
/>,
                <user
                  deleteUser={[Function]}
                  firstName="Charles"
                  lastName="Morris"
                  profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                  userId={5}
/>,
                <user
                  deleteUser={[Function]}
                  firstName="Tracey"
                  lastName="Ramos"
                  profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                  userId={6}
/>,
              ],
              "className": "users-box",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "4",
                "nodeType": "function",
                "props": Object {
                  "deleteUser": [Function],
                  "firstName": "Eve",
                  "lastName": "Holt",
                  "profileAvatar": "https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg",
                  "userId": 4,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "5",
                "nodeType": "function",
                "props": Object {
                  "deleteUser": [Function],
                  "firstName": "Charles",
                  "lastName": "Morris",
                  "profileAvatar": "https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg",
                  "userId": 5,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "6",
                "nodeType": "function",
                "props": Object {
                  "deleteUser": [Function],
                  "firstName": "Tracey",
                  "lastName": "Ramos",
                  "profileAvatar": "https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg",
                  "userId": 6,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "ul",
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Pearson User Management
</h1>,
          <div
            className="users-list-box"
>
            <div
                        className="inner-user-box"
            >
                        <ul
                                    className="users-box"
                        >
                                    <user
                                                deleteUser={[Function]}
                                                firstName="Eve"
                                                lastName="Holt"
                                                profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                                                userId={4}
                                    />
                                    <user
                                                deleteUser={[Function]}
                                                firstName="Charles"
                                                lastName="Morris"
                                                profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                                                userId={5}
                                    />
                                    <user
                                                deleteUser={[Function]}
                                                firstName="Tracey"
                                                lastName="Ramos"
                                                profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                                                userId={6}
                                    />
                        </ul>
            </div>
</div>,
        ],
        "className": "pearon-users",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Pearson User Management",
          },
          "ref": null,
          "rendered": "Pearson User Management",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="inner-user-box"
>
              <ul
                            className="users-box"
              >
                            <user
                                          deleteUser={[Function]}
                                          firstName="Eve"
                                          lastName="Holt"
                                          profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                                          userId={4}
                            />
                            <user
                                          deleteUser={[Function]}
                                          firstName="Charles"
                                          lastName="Morris"
                                          profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                                          userId={5}
                            />
                            <user
                                          deleteUser={[Function]}
                                          firstName="Tracey"
                                          lastName="Ramos"
                                          profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                                          userId={6}
                            />
              </ul>
</div>,
            "className": "users-list-box",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ul
                className="users-box"
>
                <user
                                deleteUser={[Function]}
                                firstName="Eve"
                                lastName="Holt"
                                profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                                userId={4}
                />
                <user
                                deleteUser={[Function]}
                                firstName="Charles"
                                lastName="Morris"
                                profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                                userId={5}
                />
                <user
                                deleteUser={[Function]}
                                firstName="Tracey"
                                lastName="Ramos"
                                profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                                userId={6}
                />
</ul>,
              "className": "inner-user-box",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <user
                    deleteUser={[Function]}
                    firstName="Eve"
                    lastName="Holt"
                    profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                    userId={4}
/>,
                  <user
                    deleteUser={[Function]}
                    firstName="Charles"
                    lastName="Morris"
                    profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                    userId={5}
/>,
                  <user
                    deleteUser={[Function]}
                    firstName="Tracey"
                    lastName="Ramos"
                    profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                    userId={6}
/>,
                ],
                "className": "users-box",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "4",
                  "nodeType": "function",
                  "props": Object {
                    "deleteUser": [Function],
                    "firstName": "Eve",
                    "lastName": "Holt",
                    "profileAvatar": "https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg",
                    "userId": 4,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "5",
                  "nodeType": "function",
                  "props": Object {
                    "deleteUser": [Function],
                    "firstName": "Charles",
                    "lastName": "Morris",
                    "profileAvatar": "https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg",
                    "userId": 5,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "6",
                  "nodeType": "function",
                  "props": Object {
                    "deleteUser": [Function],
                    "firstName": "Tracey",
                    "lastName": "Ramos",
                    "profileAvatar": "https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg",
                    "userId": 6,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "ul",
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Users should render with required props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Users />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Pearson User Management
</h1>,
        <div
          className="users-list-box"
>
          <div
                    className="inner-user-box"
          >
                    <ul
                              className="users-box"
                    >
                              <user
                                        deleteUser={[Function]}
                                        firstName="Eve"
                                        lastName="Holt"
                                        profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                                        userId={4}
                              />
                              <user
                                        deleteUser={[Function]}
                                        firstName="Charles"
                                        lastName="Morris"
                                        profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                                        userId={5}
                              />
                              <user
                                        deleteUser={[Function]}
                                        firstName="Tracey"
                                        lastName="Ramos"
                                        profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                                        userId={6}
                              />
                    </ul>
          </div>
</div>,
      ],
      "className": "pearon-users",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Pearson User Management",
        },
        "ref": null,
        "rendered": "Pearson User Management",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="inner-user-box"
>
            <ul
                        className="users-box"
            >
                        <user
                                    deleteUser={[Function]}
                                    firstName="Eve"
                                    lastName="Holt"
                                    profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                                    userId={4}
                        />
                        <user
                                    deleteUser={[Function]}
                                    firstName="Charles"
                                    lastName="Morris"
                                    profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                                    userId={5}
                        />
                        <user
                                    deleteUser={[Function]}
                                    firstName="Tracey"
                                    lastName="Ramos"
                                    profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                                    userId={6}
                        />
            </ul>
</div>,
          "className": "users-list-box",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="users-box"
>
              <user
                            deleteUser={[Function]}
                            firstName="Eve"
                            lastName="Holt"
                            profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                            userId={4}
              />
              <user
                            deleteUser={[Function]}
                            firstName="Charles"
                            lastName="Morris"
                            profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                            userId={5}
              />
              <user
                            deleteUser={[Function]}
                            firstName="Tracey"
                            lastName="Ramos"
                            profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                            userId={6}
              />
</ul>,
            "className": "inner-user-box",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <user
                  deleteUser={[Function]}
                  firstName="Eve"
                  lastName="Holt"
                  profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                  userId={4}
/>,
                <user
                  deleteUser={[Function]}
                  firstName="Charles"
                  lastName="Morris"
                  profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                  userId={5}
/>,
                <user
                  deleteUser={[Function]}
                  firstName="Tracey"
                  lastName="Ramos"
                  profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                  userId={6}
/>,
              ],
              "className": "users-box",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "4",
                "nodeType": "function",
                "props": Object {
                  "deleteUser": [Function],
                  "firstName": "Eve",
                  "lastName": "Holt",
                  "profileAvatar": "https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg",
                  "userId": 4,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "5",
                "nodeType": "function",
                "props": Object {
                  "deleteUser": [Function],
                  "firstName": "Charles",
                  "lastName": "Morris",
                  "profileAvatar": "https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg",
                  "userId": 5,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "6",
                "nodeType": "function",
                "props": Object {
                  "deleteUser": [Function],
                  "firstName": "Tracey",
                  "lastName": "Ramos",
                  "profileAvatar": "https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg",
                  "userId": 6,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "ul",
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Pearson User Management
</h1>,
          <div
            className="users-list-box"
>
            <div
                        className="inner-user-box"
            >
                        <ul
                                    className="users-box"
                        >
                                    <user
                                                deleteUser={[Function]}
                                                firstName="Eve"
                                                lastName="Holt"
                                                profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                                                userId={4}
                                    />
                                    <user
                                                deleteUser={[Function]}
                                                firstName="Charles"
                                                lastName="Morris"
                                                profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                                                userId={5}
                                    />
                                    <user
                                                deleteUser={[Function]}
                                                firstName="Tracey"
                                                lastName="Ramos"
                                                profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                                                userId={6}
                                    />
                        </ul>
            </div>
</div>,
        ],
        "className": "pearon-users",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Pearson User Management",
          },
          "ref": null,
          "rendered": "Pearson User Management",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="inner-user-box"
>
              <ul
                            className="users-box"
              >
                            <user
                                          deleteUser={[Function]}
                                          firstName="Eve"
                                          lastName="Holt"
                                          profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                                          userId={4}
                            />
                            <user
                                          deleteUser={[Function]}
                                          firstName="Charles"
                                          lastName="Morris"
                                          profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                                          userId={5}
                            />
                            <user
                                          deleteUser={[Function]}
                                          firstName="Tracey"
                                          lastName="Ramos"
                                          profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                                          userId={6}
                            />
              </ul>
</div>,
            "className": "users-list-box",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ul
                className="users-box"
>
                <user
                                deleteUser={[Function]}
                                firstName="Eve"
                                lastName="Holt"
                                profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                                userId={4}
                />
                <user
                                deleteUser={[Function]}
                                firstName="Charles"
                                lastName="Morris"
                                profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                                userId={5}
                />
                <user
                                deleteUser={[Function]}
                                firstName="Tracey"
                                lastName="Ramos"
                                profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                                userId={6}
                />
</ul>,
              "className": "inner-user-box",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <user
                    deleteUser={[Function]}
                    firstName="Eve"
                    lastName="Holt"
                    profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                    userId={4}
/>,
                  <user
                    deleteUser={[Function]}
                    firstName="Charles"
                    lastName="Morris"
                    profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                    userId={5}
/>,
                  <user
                    deleteUser={[Function]}
                    firstName="Tracey"
                    lastName="Ramos"
                    profileAvatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                    userId={6}
/>,
                ],
                "className": "users-box",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "4",
                  "nodeType": "function",
                  "props": Object {
                    "deleteUser": [Function],
                    "firstName": "Eve",
                    "lastName": "Holt",
                    "profileAvatar": "https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg",
                    "userId": 4,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "5",
                  "nodeType": "function",
                  "props": Object {
                    "deleteUser": [Function],
                    "firstName": "Charles",
                    "lastName": "Morris",
                    "profileAvatar": "https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg",
                    "userId": 5,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "6",
                  "nodeType": "function",
                  "props": Object {
                    "deleteUser": [Function],
                    "firstName": "Tracey",
                    "lastName": "Ramos",
                    "profileAvatar": "https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg",
                    "userId": 6,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "ul",
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
